### PR TITLE
Lux 128

### PR DESCRIPTION
**Pull request by emmirose**

### CODE REVIEW

File: src/main/java/com/luxiergerie/Controllers/PurchaseController.java [hunk 1]
```diff
@@ -113,10 +113,12 @@ public PurchaseDTO getPurchase(@PathVariable("id") UUID purchaseId) {
 
   @PostMapping("")
   public PurchaseDTO createPurchase(@RequestBody PurchaseDTO purchaseDTO) {
-    Optional<Purchase> purchaseOptional = purchaseRepository.findById(purchaseDTO.getId());
-    if(purchaseOptional.isPresent()){
-      throw new ResponseStatusException(HttpStatus.CONFLICT, "Purchase already exists with ID: " + purchaseDTO.getId());
-    }
+    if(purchaseDTO.getId() != null) {
```
Comment (JocelynDeloose) [inline src/main/java/com/luxiergerie/Controllers/PurchaseController.java]: Object ... isNotNull()

File: src/main/java/com/luxiergerie/Controllers/PurchaseController.java [hunk 2]
```diff
@@ -145,4 +148,4 @@ public void deletePurchase(@PathVariable("id") UUID id) {
     }
     purchaseRepository.deleteById(id);
   }
```
Comment (JocelynDeloose) [inline src/main/java/com/luxiergerie/Controllers/PurchaseController.java:150]: espace

File: src/main/java/com/luxiergerie/Domain/Entity/Client.java [hunk 3]
```diff
@@ -45,12 +45,15 @@ public class Client {
...
     @Column(name = "authority")
-    private List<String> authorities;
+    private List<String> authorities = new ArrayList<>();
 
     @OneToMany(mappedBy = "client")
-    private List<Sojourn> sojourns;
+    private List<Sojourn> sojourns = new ArrayList<>();
 
     public Collection<? extends GrantedAuthority> getAuthorities() {
+      if (this.authorities == null) {
```
Comment (JocelynDeloose) [inline src/main/java/com/luxiergerie/Domain/Entity/Client.java]: isNull

**Review by JocelynDeloose — CHANGES_REQUESTED** [review]

**Review by JocelynDeloose — APPROVED** [review]

**Review by mohamedehn — APPROVED** [review]
Good job

**Pull request merged**